common4j/src/main/com/microsoft/identity/common/java/providers/microsoft/microsoftsts/MicrosoftStsOAuth2Strategy.java

```diff
@@ -608,6 +611,12 @@ protected TokenResult getTokenResultFromHttpResponse(@NonNull final HttpResponse
                     tokenResponse.setCliTelemSubErrorCode(cliTelemInfo.getServerSubErrorCode());
                 }
             }
+
+            if (null != responseHeaders.get(XMS_CCS_REQUEST_ID)) {
+                Span.current().setAttribute(
+                        AttributeName.ccs_request_id.name(),
+                        responseHeaders.get(XMS_CCS_REQUEST_ID).get(0));
```

Comment:
I think @rpdome found that this way of accessing the headers is unsafe and he created a new method for it I believe. @rpdome Can you share that here?